### PR TITLE
QT: Implement 'check for update' dialog, download and install option

### PR DESCRIPTION
**Pull request by effectsToCause**

### CODE REVIEW

File: src/qt/res/style.qss [hunk 1]
```diff
@@ -424,18 +424,42 @@ QWidget#bgWidget {
...
-QDialog#AskPassphraseDialog {
+/** Dialogs **/
+QDialog{
     border-radius: 5px;
     background-color: white;
 }
 
-QDialog#AskPassphraseDialog QLabel,
-QDialog#AskPassphraseDialog QCheckBox#toggleShowPasswordButton {
+QDialog QLabel,
```
Comment (mderasse) [inline src/qt/res/style.qss]: Can you keep them more specific ? you will impact some part of the send coin by doing so ;)

File: src/qt/updatedialog.cpp [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
     setWindowTitle(tr("Check for Update"));
+    ui->updateButton->setVisible(false);
+    ui->progressBar->setVisible(false);
+    downloadVersionFile();
+    clientName = getUpdatedClient();
+    if (clientName.compare(" ") != 0){
+
+        ui->updateButton->setVisible(true);
+        std::string stdstr = "Update is available! Click update to download " +clientName;
+        QString qstr = QString::fromStdString(stdstr);
```
Comment (mderasse) [inline src/qt/updatedialog.cpp]:
```suggestion
        QString qstr = QString(QObject::tr("Update is available! Click update to download %1")).arg(clientName);
```
Like that we could translate :)

**Review by mderasse — CHANGES_REQUESTED** [review]

File: src/qt/res/style.qss (hunk 1 above)
Comment (effectsToCause) [reply src/qt/res/style.qss]: yes sure

File: src/qt/updatedialog.cpp (hunk 2 above)
Comment (mderasse) [reply src/qt/updatedialog.cpp]: ?

Comment (effectsToCause) [reply src/qt/updatedialog.cpp]: thought this was committed my bad, it is now

**Review by mderasse — APPROVED** [review]

**Pull request merged**